### DISCUSSION TITLE
Wal-g backups on free tier, replication connection

### DISCUSSION THREAD

**Discussion by snorremd** [supabase/supabase#9272] [Questions]
Hi!

I`m trying to set up wal-g backups for one of my servers. I currently have the following config/script. According to the documentation wal-g is one of two recommended self-hosted backup solutions for the free tier: https://supabase.com/docs/going-into-prod#availability

```sh
# Settings related to S3 bucket
export AWS_ENDPOINT=s3.<host>
export AWS_ACCESS_KEY_ID=<id>
export AWS_SECRET_ACCESS_KEY=<id>

# Database connection settings
export PGHOST="db.<host>.supabase.co"
export PGPORT=5432
export PGDATABASE=postgres
export PGUSER=postgres
export PGPASSWORD="<password>"

# Settings related to walg
export WALG_S3_PREFIX=s3://<bucket>
export WALG_COMPRESSION_METHOD=brotli
export WALG_S3_STORAGE_CLASS=ONEZONE_IA

# Code to make backup happen
wal-g backup-push

```

When running the script I get the following error:

```sh
INFO: 2022/10/01 15:05:48.288083 Running remote backup through Postgres connection.
INFO: 2022/10/01 15:05:48.288211 Features like delta backup are disabled, there might be a performance impact.
INFO: 2022/10/01 15:05:48.288224 To run with local backup functionalities, supply [db_directory].
ERROR: 2022/10/01 15:05:48.451979 failed to connect to `host=db.<host>.supabase.co user=postgres database=postgres`: server error (FATAL: no pg_hba.conf entry for replication connection from host "<ip>", user "postgres", SSL encryption (SQLSTATE 28000))
```

It seems to me there might be something needed from Supabase`s side to support (remote) wal-g connections?

**Comment by michaelwills** [comment]
@snorremd that link seems to be broken now. I do see this blog post from a couple years back:

[Continuous PostgreSQL Backups using WAL-G](https://supabase.com/blog/continuous-postgresql-backup-walg)

If that _is_ possible though that opens up possibilities.

**Comment by michaelwills** [comment]
I see what you mean:

[Production Readiness | Supabase](https://web.archive.org/web/20221027100922/https://supabase.com/docs/going-into-prod)

So it at least was a suggested option at one point.

**Reply by michaelwills** [reply]
@snorremd so it looks like we can configure streaming replication:

https://github.com/supabase/supabase/discussions/10119

however that it look like you can set up archival files wal-g would use. Still digging on that one.

**Reply by michaelwills** [reply]
@w3b6x9 pointed me to this article:

https://supabase.com/blog/continuous-postgresql-backup-walg
